ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -1,35 +1,46 @@
 // Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-package com.daml.ledger.participant.state.kvutils.committing
+package com.daml.ledger.participant.state.kvutils.committer
 
 import java.time.Instant
 
 import com.codahale.metrics.Counter
 import com.daml.ledger.participant.state.kvutils.Conversions._
 import com.daml.ledger.participant.state.kvutils.DamlKvutils._
-import com.daml.ledger.participant.state.kvutils.committing.Common.Commit._
-import com.daml.ledger.participant.state.kvutils.committing.Common._
-import com.daml.ledger.participant.state.kvutils.committing.ProcessTransactionSubmission._
+import com.daml.ledger.participant.state.kvutils.committer.Committer._
 import com.daml.ledger.participant.state.kvutils.{Conversions, DamlStateMap, Err, InputsAndEffects}
-import com.daml.ledger.participant.state.v1.{Configuration, ParticipantId, RejectionReason}
+import com.daml.ledger.participant.state.v1.{Configuration, RejectionReason}
 import com.daml.lf.archive.Decode
 import com.daml.lf.archive.Reader.ParseError
 import com.daml.lf.crypto
 import com.daml.lf.data.Ref.{PackageId, Party}
 import com.daml.lf.data.Time.Timestamp
 import com.daml.lf.engine.{Blinding, Engine}
 import com.daml.lf.language.Ast
-import com.daml.lf.transaction.Transaction.AbsTransaction
 import com.daml.lf.transaction.{BlindingInfo, GenTransaction, Node}
+import com.daml.lf.transaction.Transaction.AbsTransaction
 import com.daml.lf.value.Value
 import com.daml.lf.value.Value.AbsoluteContractId
 import com.daml.metrics.Metrics
 import com.google.protobuf.{Timestamp => ProtoTimestamp}
-import org.slf4j.{Logger, LoggerFactory}
 
 import scala.collection.JavaConverters._
 
+private[kvutils] object TransactionCommitter {
+
+  case class Digest(submission: DamlTransactionEntry) {
+    val ledgerEffectiveTime: Timestamp = parseTimestamp(submission.getLedgerEffectiveTime)
+    val submitterInfo: DamlSubmitterInfo = submission.getSubmitterInfo
+    val commandId: String = submitterInfo.getCommandId
+    val submitter: Party = Party.assertFromString(submitterInfo.getSubmitter)
+    lazy val abs: AbsTransaction = Conversions.decodeTransaction(submission.getTransaction)
```

Comment:
```suggestion
    lazy val absoluteTransaction: AbsTransaction = Conversions.decodeTransaction(submission.getTransaction)
```

Reply:
Why does it help for it to be `lazy`?

Reply:
That was there in the original implementation. I am guessing it is to avoid decoding upfront, when the transaction can fail on a more rudimentary validation before